witness/golang/omniwitness/usbarmory/storage/slots/journal_test.go

```diff
@@ -0,0 +1,231 @@
+package slots
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/trillian-examples/witness/golang/omniwitness/usbarmory/storage/testonly"
+)
+
+func magic0Hdr() [4]byte {
+	return [4]byte{magic0[0], magic0[1], magic0[2], magic0[3]}
+}
+
+func TestEntryMarshal(t *testing.T) {
+	for _, test := range []struct {
+		name    string
+		e       *entry
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "golden",
+			e: &entry{
+				Magic:      magic0Hdr(),
+				Revision:   42,
+				DataLen:    uint64(len("hello")),
+				DataSHA256: sha256.Sum256([]byte("hello")),
+				Data:       []byte("hello"),
+			},
+			want: []byte{
+				'T', 'F', 'J', '0',
+				0x00, 0x00, 0x00, 0x2a,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x05,
+				0x2c, 0xf2, 0x4d, 0xba, 0x5f, 0xb0, 0xa3, 0x0e, 0x26, 0xe8, 0x3b, 0x2a, 0xc5, 0xb9, 0xe2, 0x9e, 0x1b, 0x16, 0x1e, 0x5c, 0x1f, 0xa7, 0x42, 0x5e, 0x73, 0x04, 0x33, 0x62, 0x93, 0x8b, 0x98, 0x24,
+				'h', 'e', 'l', 'l', 'o',
+			},
+		}, {
+			name: "bad magic",
+			e: &entry{
+				Magic:      [4]byte{'n', 'O', 'p', 'E'},
+				Revision:   42,
+				DataLen:    uint64(len("hello")),
+				DataSHA256: sha256.Sum256([]byte("hello")),
+				Data:       []byte("hello"),
+			},
+			wantErr: true,
+		}, {
+			name: "bad hash",
+			e: &entry{
+				Magic:      magic0Hdr(),
+				Revision:   42,
+				DataLen:    uint64(len("hello")),
```

Comment:
I'm not sure what the expected result should be, though.
e.g. it _could_ say _"this record, in this journal geometry, should not have been allowed to have been written since it's too large"_, but do we want that behaviour? 
(Actually, it couldn't because `unmarshalEntry` doesn't have access to the journal geometry, but the same question about behaviour is true for any arbitrary upper limit on allowable entry data size.)